packages/wdio-browserstack-service/src/Percy/Percy-Handler.ts

```diff
@@ -0,0 +1,88 @@
+import type { Capabilities } from '@wdio/types'
+import type { BeforeCommandArgs, AfterCommandArgs } from '@wdio/reporter'
+
+import {
+    o11yClassErrorHandler
+} from '../util.js'
+import PercyCaptureMap from './PercyCaptureMap.js'
+
+import * as PercySDK from './PercySDK.js'
+
+class _PercyHandler {
+    private _testMetadata: { [key: string]: any } = {}
+    private sessionName?: string
+    private _isAppAutomate?: boolean
+    public _percyScreenshotCounter: any = 0
+
+    constructor (
+        private _percyAutoCaptureMode: string | undefined,
+        private _browser: WebdriverIO.Browser | WebdriverIO.MultiRemoteBrowser,
+        private _capabilities: Capabilities.RemoteCapability,
+        isAppAutomate?: boolean,
+        private _framework?: string
+    ) {
+        this._isAppAutomate = isAppAutomate
+    }
+
+    _setSessionName(name: string) {
+        this.sessionName = name
+    }
+
+    async teardown () {
+        await new Promise<void>((resolve) => {
+            setInterval(() => {
```

Comment:
Shouldn't this be `setTimeout`? Do we want to run this every one second?

Reply:
Yes the promise is resolved as soon as this counter goes to 0 i.e no more screenshots in processing stage.